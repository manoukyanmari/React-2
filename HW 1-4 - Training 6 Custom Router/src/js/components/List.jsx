// src/js/components/List.jsx
import React, {Component} from "react";
import { connect } from "react-redux";
import {Router, Route, handleClick} from '../App.jsx';
import 'babel-polyfill';

const mapStateToProps = state => {
    return {
        movies: state.movies
    };
};


//dumb component
class List extends Component {
    constructor(props) {
        super(props);
    };

    render() {
            const ConnectedList = ({ movies }) => (
                    <div className="row mt-5">
                        <div className="container">
                            {movies.map(el => (
                                <div className="col-12 col-sm-4 col-md-3 p-2" key={el.id}>
                                    <div className="text-center border height100">
                                        <div>
                                            <Link to="/movie">
                                                <img src={el.src}/>
                                            </Link>
                                        </div>
                                        <h5>{el.title}</h5>
                                    </div>
                                </div>
                            ))}
                        </div>
                    </div>
            );
            const ListComponent = connect(mapStateToProps)(ConnectedList);
            return (
                <Router>
                    const {handleClick} = context;
                    <Route path="/movie" Component={ListComponent}/>
                    {/*<ListComponent/>*/}
                </Router>
                )

    }

}


export default List;

