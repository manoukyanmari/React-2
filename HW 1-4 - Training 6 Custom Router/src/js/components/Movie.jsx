import React from "react";

const marginStyle = {
    marginTop: '30px'
};

class Movie extends React.Component {
    render() {
        return (
            <div className="container">
                <div className="jumbotron col-md-12">
                    <div className="col-md-4">
                        <img src="../../../src/imgs/machinist.JPG"/>
                    </div>
                    <div className="col-md-8">
                        <div className="row">
                            <h2>Machinist</h2>
                            <div className="row">
                                <label htmlFor="title">Oscar-winning Movies</label>
                            </div>
                            <div className="row">
                                <div className="col-md-3 text-left">1994</div>
                                <div className="col-md-3 text-right">154 min</div>
                            </div>
                        </div>
                        <div className="row" style={marginStyle}>
                            Trevor Reznik (Christian Bale) is a machinist whose insomnia has led to his becoming emaciated. His appearance and behavior keep his coworkers away, and they eventually turn against him when he is involved in an accident which causes his coworker, Miller, to lose his left arm. Trevor, who was distracted by an unfamiliar co-worker named Ivan, is blamed for the accident. No one at the factory knows of Ivan and there are no records of him. Trevor seems to find peace only with Stevie, a prostitute with genuine affection for him, and with Maria, a waitress at an airport diner he frequents. He is haunted by brief flashes of recurring imagery, and things such as his car cigarette lighter take on a menacing air. A mysterious series of post-it notes appear on his refrigerator, depicting a game of hangman.
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
export default Movie;