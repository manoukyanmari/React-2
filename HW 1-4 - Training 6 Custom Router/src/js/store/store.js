// src/js/store/store.js
import { createStore } from "redux";
import rootReducer from "../reducers/reducer";
import {addMovie} from "../actions/action";

const store = createStore(rootReducer);
store.subscribe(() => console.log('Look ma, Redux!!'));
store.dispatch( addMovie({ title: 'React Redux Tutorial for Beginners', id: 1, src: '../../../src/imgs/machinist.JPG' }));
export default store;