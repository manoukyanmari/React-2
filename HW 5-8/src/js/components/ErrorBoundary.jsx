import React, { Component } from 'react';

class ErrorBoundary extends Component {
    constructor(props) {
        super(props);
        this.state = {
            hasError: false,
            logErrorToMyService: function(error, info) {
                console.log(error, 'error');
            }
        };
    }

    componentDidCatch(error, info) {
        this.setState({ hasError: true });
        // You can also log the error to an error reporting service
        this.state.logErrorToMyService(error, info);
    }

    render() {
        if (this.state.hasError) {
            return <h1>Uh oh! Something went wrong.</h1>;
        }
        return this.props.children;
    }
}

export default ErrorBoundary;