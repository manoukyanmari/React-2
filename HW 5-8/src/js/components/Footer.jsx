import React from "react";
const Footer = (props) => {
    return(
        <section id="footer" className="footer">
            <div className="container">
                <div className="row text-sm-left text-md-left">
                    <div>
                        <h6>Footer Information</h6>
                    </div>
                </div>
            </div>
        </section>
    )
};
export default Footer;