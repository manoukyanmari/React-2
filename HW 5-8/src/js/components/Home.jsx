import React from "react";
import Form from "./form/Form.jsx";

class Home extends React.Component {
    render() {
        return (
            <div className="container">
                <div className="jumbotron">
                <h2>NetFlix roulette</h2>
                <label htmlFor="title">FIND YOUR MOVIE</label>
                    <Form />
                </div>
            </div>
        )
    }
}
export default Home;