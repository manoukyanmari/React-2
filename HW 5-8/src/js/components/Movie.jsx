import React, {Component} from "react";
import {ROOT_URL} from "../constants/action-types";
import Moment from 'moment';

const marginStyle = {
    marginTop: '30px'
};

const imgStyle = {
    maxWidth: '80%'
};

class Movie extends Component {
    // componentDidMount() {
    //     console.log(this.props);
    //     const { match: { params } } = this.props;
    //
    //     axios.get(`/api/users/${params.id}`)
    //         .then(({ data: movie }) => {
    //             console.log('user', movie);
    //
    //             this.setState({ movie });
    //         });
    // }
    constructor(props) {
        super(props);
            this.state = {movie: {}};
            fetch(`${ROOT_URL}movies/${props.match.params.id}`)
                .then(res => res.json())
                .then(
                    (result) => {
                        this.setState({movie:result});

                    },
                    (error) => {
                        this.setState({
                            isLoaded: true,
                            error
                        });
                    }
                );
    };

    render() {
        const {movie} = this.state;
        return (
            <div className="container">
                <div className="jumbotron col-md-12">
                    <div className="col-md-4">
                        <img style={imgStyle} src={movie.poster_path}/>
                    </div>
                    <div className="col-md-8">
                        <div className="row">
                            <h2>{movie.title}</h2>
                            <div>
                                <span>Genre: </span>
                                <label htmlFor="title">
                                    {movie.genres? movie.genres.map(function(name, index){
                                        return <span key={ index }>{name}, </span>;
                                    }) : ''}
                                    </label>
                            </div>
                            <div className="row">
                                <div className="col-md-3 text-left">{Moment(movie.release_date).format('YYYY-MM-DD')}</div>
                                <div className="col-md-3 text-right">{movie.runtime}</div>
                            </div>
                        </div>
                        <div className="row" style={marginStyle}>
                            {movie.overview}
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
export default Movie;