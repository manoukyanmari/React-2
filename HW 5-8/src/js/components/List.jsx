// src/js/components/List.jsx
import React, {Component} from "react";
import { connect } from "react-redux";
import 'babel-polyfill';
import {BrowserRouter as Router, Route, Link, Switch} from "react-router-dom";
import Movie from "./Movie.jsx";
import Root from "./../App.jsx";

const mapStateToProps = state => {
    return {
        articles: state.articles
    };
};


//dumb component
class List extends Component {
    constructor(props) {
        super(props);
        console.log(props,'aaaa');
    };

    componentDidMount() {
        console.log(this.props.location.search) // "?filter=top&origin=im"
    }
    render() {
            const ConnectedList = ({ articles }) => (
                <div className="row mt-5">
                    <div className="container">
                        <Route path={this.props.match ? this.props.match.url + '/movie/:id' : ''} component={Movie}/>
                        {articles.map(el => (
                            <div className="col-12 col-sm-4 col-md-3 p-2" key={el.id}>
                                <div className="text-center border height100">
                                    <div>
                                        <Link to={this.props.match.url + '/movie/' + el.id}>
                                            <img src={el.src}/>
                                        </Link>
                                    </div>
                                    <h5>{el.title}</h5>
                                </div>
                            </div>
                        ))}
                    </div>
                </div>

            );
            const ListComponent = connect(mapStateToProps)(ConnectedList);
            return (
                    <ListComponent/>
                )

    }

}


export default List;

