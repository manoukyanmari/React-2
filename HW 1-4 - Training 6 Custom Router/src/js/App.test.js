import React from "react";
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import ErrorBoundary from "./components/ErrorBoundary.jsx";
import { Switch } from 'react-router'
import { Route, Link, BrowserRouter as Router } from 'react-router-dom'
import List from "./components/List.jsx";
import Home from "./components/Home.jsx";
import Movie from "./components/Movie.jsx";
import Footer from "./components/Footer.jsx";
import ReactDOM from "react-dom";
import { Provider } from 'react-redux'
import store from "./store/store";


describe('There is an initial component and error test',()=>{
    let wrapper;
    let componentWrapper;
    let footer;
    let list;
    let home;
    let movie;
    let provider;

    beforeEach(()=>{
        wrapper = shallow(<ErrorBoundary/>)
    });

    it('+++ render the DUMB component', () => {
        expect(wrapper.length).toEqual(1)
    });

    it('renders correctly the store for provider', () => {
        provider = shallow(<Provider store={store}/>);
        expect(provider.length).toEqual(1)
    });
    it('renders correctly the Movie component', () => {
        movie = shallow(<Movie component={movie}/>);
        expect(movie.length).toEqual(1)
    });
    it('renders correctly Footer component', () => {
        footer = shallow(<Footer/>);
        expect(footer.length).toEqual(1)
    });
    it('renders correctly the List component', () => {
        list = shallow(<List/>);
        expect(list.length).toEqual(1)
    });
    it('renders correctly the Home Component', () => {
        home = shallow(<Home/>);
        expect(home.length).toEqual(1)
    });
});