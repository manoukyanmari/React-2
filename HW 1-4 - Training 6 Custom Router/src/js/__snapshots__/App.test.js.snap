// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`There is an initial component and error test renders correctly the store for provider 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeState": Object {
        "articles": Array [
          Object {
            "id": 1,
            "src": "../../../src/imgs/machinist.JPG",
            "title": "React Redux Tutorial for Beginners",
          },
        ],
      },
    }
  }
/>
`;
